src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,249 @@
+          setSelectedOption(EMPTY_OPTION);
+          setInputValue(EMPTY_OPTION.name);
+          onChange(EMPTY_OPTION.value);
+        } else {
+          await fetchSelectedOption(selectionValue as string);
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        if (inputValue === "") {
+          setOptions([]);
+          setSelectedOption(null);
+          onChange("");
+        } else {
+          const searchParams = new URLSearchParams();
+          const organization_id = organizationId
+            ? organizationId
+            : await getOrganizationId(userId);
+          searchParams.set("name", inputValue.toLocaleLowerCase());
+          searchParams.set("source", "zen:ticket");
+          searchParams.set("field_id", fieldId.toString());
+          searchParams.set("user_id", userId.toString());
+          searchParams.set("organization_id", organization_id);
+          setIsLoadingOptions(true);
+          try {
+            const response = await fetch(
+              `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+            );
+
+            const data = await response.json();
+            if (data !== undefined) {
+              setOptions(
+                data.custom_object_records.map(
+                  ({ name, id }: { name: string; id: string }) => ({
+                    name,
+                    value: id,
+                  })
+                )
+              );
+            } else {
+              setOptions([]);
+            }
+          } catch (error) {
+            return error;
+          } finally {
+            setIsLoadingOptions(false);
+          }
+        }
+      }
+      return;
+    },
+    [
+      customObjectKey,
+      fetchSelectedOption,
+      userId,
+      organizationId,
+      fieldId,
+      onChange,
+    ]
+  );
+
+  const debounceHandleChange = useMemo(() => debounce(handleChange, 300), []);
+
+  useEffect(() => {
+    return () => debounceHandleChange.cancel();
+  }, [debounceHandleChange]);
+
+  useEffect(() => {
+    if (value && !options.find((option) => option.value === value)) {
+      fetchSelectedOption(value as string);
+      return;
+    }
+  }, []);
+
+  const onFocus = () => {
+    setInputValue("");
+  };
+
+  return (
+    <GardenField>
+      <Label>
+        {label}
+        {required && <Span aria-hidden="true">*</Span>}
+      </Label>
+      {description && (
+        <Hint dangerouslySetInnerHTML={{ __html: description }} />
+      )}
+      <input type="hidden" name={name} value={selectedOption?.value} />
+      <Combobox
+        inputProps={{ required }}
+        startIcon={<SearchIcon />}
+        validation={error ? "error" : undefined}
+        inputValue={inputValue}
+        selectionValue={selectedOption?.value}
+        onFocus={onFocus}
+        onChange={debounceHandleChange}
+        renderValue={({ selection }) =>
```

Comment:
When the form has an error and the page is reloaded I don't see the selected record name in the field. I think we should render `selectedOption?.name` and not `selection?.label`, because in this case `selection` is null

Reply:
Changed